pkg/pyramid/tier_fs.go

```diff
@@ -85,7 +81,13 @@ func (tfs *TierFS) log(ctx context.Context) logging.Logger {
 //  2. Remove workspace directories and all its content if it
 //     exists under the namespace dir.
 func (tfs *TierFS) handleExistingFiles() error {
-	if err := filepath.Walk(tfs.fsLocalBaseDir, func(p string, info os.FileInfo, err error) error {
+	// setup cache dir location with separator as suffix. store are relative without prefix.
+	dir := tfs.fsLocalBaseDir
+	if !strings.HasSuffix(dir, string(filepath.Separator)) {
+		dir += string(filepath.Separator)
```

Comment:
`filepath.Separator` might not be `/`, but we always do want that slash!

```suggestion
	if !strings.HasSuffix(dir, "/") {
		dir += "/"